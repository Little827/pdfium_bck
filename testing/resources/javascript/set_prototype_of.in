{{header}}
{{object 1 0}} <<
  /Type /Catalog
  /Pages 2 0 R
  /OpenAction 10 0 R
>>
endobj
{{object 2 0}} <<
  /Type /Pages
  /Count 1
  /Kids [
    3 0 R
  ]
>>
endobj
% Page number 0.
{{object 3 0}} <<
  /Type /Page
  /Parent 2 0 R
  /Resources <<
    /Font <</F1 15 0 R>>
  >>
  /Contents [21 0 R]
  /MediaBox [0 0 612 792]
>>
% OpenAction action
{{object 10 0}} <<
  /Type /Action
  /S /JavaScript
  /JS 11 0 R
>>
endobj
% JS program to exexute
{{object 11 0}} <<
  {{streamlen}}
>>
stream
function testSetPrototypeOf(name) {
  var that;
  try {
    that = eval(name);
  } catch (e) {
    app.alert("FAIL: No such " + name);
    return;
  }
  try {
    Object.setPrototypeOf(that, {});
    app.alert("FAIL: setPrototypeOf(" + name + "): returned");
  } catch (e) {
    app.alert("PASS: setPrototypeOf(" + name + "): " + e);
  }
}
function doTest() {
  if (typeof [].__proto__ === "undefined" &&
      typeof Object.setPrototypeOf === "undefined") {
    app.alert("MOOT: JS doesn't contain setPrototypeOf() or __proto__");
    return;
  }
  testSetPrototypeOf("this");
  testSetPrototypeOf("app");
  testSetPrototypeOf("border");
  testSetPrototypeOf("color");
  testSetPrototypeOf("console");
  testSetPrototypeOf("display");
  testSetPrototypeOf("event");
  testSetPrototypeOf("font");
  testSetPrototypeOf("global");
  testSetPrototypeOf("highlight");
  testSetPrototypeOf("position");
  testSetPrototypeOf("scaleHow");
  testSetPrototypeOf("scaleWhen");
  testSetPrototypeOf("style");
  testSetPrototypeOf("util");
  testSetPrototypeOf("zoomtype");

  // But the usual mischief still works.
  try {
    Object.setPrototypeOf({}, Array);
    app.alert("Ordinary objects PASS");
  } catch (e) {
    app.alert("Ordinary objects FAIL" + e);
  }
}
doTest();
endstream
endobj
{{xref}}
{{trailer}}
{{startxref}}
%%EOF
