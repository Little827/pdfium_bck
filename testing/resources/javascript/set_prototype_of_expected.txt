Alert: PASS: testSetPrototypeOf(this): TypeError: Immutable prototype object '[object global]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(app): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(border): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(color): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(console): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(display): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(event): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(font): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(global): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(highlight): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(position): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(scaleHow): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(scaleWhen): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(style): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(util): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: PASS: testSetPrototypeOf(zoomtype): TypeError: Immutable prototype object '[object Object]' cannot have their prototype testSet
Alert: Ordinary objects PASS
